File: apps/web/src/types/employee.ts
export type EmployeeRole = "OWNER" | "MANAGER" | "RECEPTIONIST" | "MECHANIC" | "CLIENT";

export interface Employee {
  id: string;
  first_name: string;
  last_name: string;
  full_name: string;
  phone: string | null;
  email: string | null;
  role: EmployeeRole;
  position: string | null;
  hire_date: string | null;
  salary: number | null;
  active: boolean;
  created_at: string | null;
}

export interface EmployeeRequest {
  firstName: string;
  lastName: string;
  phone?: string;
  email?: string;
  role: EmployeeRole;
  position?: string;
  hireDate?: string;
  salary?: number;
}

File: apps/api/src/main/java/rd/tallerfacil/api/employee/domain/Employee.java
package rd.tallerfacil.api.employee.domain;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import rd.tallerfacil.api.auth.domain.RoleName;
import rd.tallerfacil.api.shared.domain.BaseEntity;

import java.math.BigDecimal;
import java.time.LocalDate;

@Entity
@Table(name = "employees")
@Getter
@Setter
@NoArgsConstructor
public class Employee extends BaseEntity {

    @Column(name = "first_name", nullable = false, length = 100)
    private String firstName;

    @Column(name = "last_name", nullable = false, length = 100)
    private String lastName;

    @Column(length = 20)
    private String phone;

    @Column(length = 150)
    private String email;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false, length = 20)
    private RoleName role = RoleName.MECHANIC;

    @Column(length = 100)
    private String position;

    @Column(name = "hire_date")
    private LocalDate hireDate;

    @Column(precision = 12, scale = 2)
    private BigDecimal salary;

    @Column(nullable = false)
    private boolean active = true;
}
